okhttp/src/main/kotlin/okhttp3/internal/platform/Android10Platform.kt

```diff
@@ -56,6 +57,14 @@ class Android10Platform : Platform() {
       // No TLS extensions if the socket class is custom.
       socketAdapters.find { it.matchesSocket(sslSocket) }?.getSelectedProtocol(sslSocket)
 
+  @SuppressLint("NewApi")
+  override fun getStackTraceForCloseable(closer: String): CloseGuard = CloseGuard().apply { open(closer) }
```

Comment:
This file is named `Android10` (API 29) but it references a class from API 30

Reply:
Doh. Will fix.

Is there a better approach now? i.e. remove this and rely on LeakCanary or similar? As a developer only tool, maybe we could drop support for the legacy mode and only support on Android 30+?

Reply:
So here's a weird observation: what do we get by using CloseGuard over the behavior in the base class?

I think when we introduced this it was attempting to take advantage of CloseGuard when a connection is leaked. But there's nothing difficult or proprietary about CloseGuard; it just hooks into the GC in a way we can do ourselves.

So I wonder... what happens if we drop all the overrides of this stuff and just rely on the base class behavior? I think it is particularly nice with the Logger.FINE option.

Reply:
1) Hook into StrictMode system

```
D/StrictMode: StrictMode policy violation: android.os.strictmode.LeakedClosableViolation: A resource was acquired at attached stack trace but never released. See java.io.Closeable for information on avoiding resource leaks.
        at android.os.StrictMode$AndroidCloseGuardReporter.report(StrictMode.java:1910)
        at dalvik.system.CloseGuard.warnIfOpen(CloseGuard.java:286)
```

2) Maybe cheaper than a stacktrace?

But... we don't use it properly https://developer.android.com/reference/android/util/CloseGuard.  We never call close, because we only call warnIfOpen when we've decided using the reference checks.

```
guard.close();
          ...;
          if (Build.VERSION.SDK_INT >= 28) {
              Reference.reachabilityFence(this);
          }
```